Content.Server/GameObjects/Components/Markers/ConditionalSpawnerComponent.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using Content.Server.GameTicking;
+using Content.Server.Interfaces.GameTicking;
+using NFluidsynth;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.Interfaces.Reflection;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using SQLitePCL;
+using Logger = Robust.Shared.Log.Logger;
+
+namespace Content.Server.GameObjects.Components.Markers
+{
+    [RegisterComponent]
+    public class ConditionalSpawnerComponent : Component, IMapInit
+    {
+        public override string Name => "ConditionalSpawner";
+
+#pragma warning disable 649
+        [Dependency] private IGameTicker _gameTicker;
+        [Dependency] private IReflectionManager _reflectionManager;
+        [Dependency] private IEntityManager _entityManager;
+        [Dependency] private IRobustRandom _robustRandom;
+#pragma warning restore 649
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public List<string> Prototypes { get; set; } = new List<string>();
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        private List<string> _gameRules = new List<string>();
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float Chance { get; set; } = 1.0f;
+
+        public List<Type> GameRules
```

Comment:
Make this return an `IEnumerable` instead by using an iterator method (`yield`)

Reply:
Done.